streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java

```diff
@@ -304,6 +363,30 @@ public void testStateStore() {
         }
     }
 
+    @Test
+    public void testSendingOldValuesNotSetIfNotMaterialized() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KTableImpl<String, String, String> table =
+            (KTableImpl<String, String, String>) builder.table("topic1", consumed);
+
+        table.enableSendingOldValues(true);
+
+        assertThat(table.sendingOldValueEnabled(), is(false));
+    }
+
+    @Test
+    public void testSendingOldValuesSetIfMaterializedForced() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KTableImpl<String, String, String> table =
+            (KTableImpl<String, String, String>) builder.table("topic1", consumed);
```

Comment:
Seems the `Matererialized.as("fred")` parameter is missing?

Reply:
Why? This test is about forced materialization. If we pass that parameter, then it would work even if not forced, right?

Reply:
Rename test to make it clear its force materialized due to sending old values, 

`shouldSetSendingOldValuesIfMaterializationForcedInternally`